Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -1376,9 +1370,29 @@ void MediaPlayerPrivateGStreamerBase::attemptToDecryptWithLocalInstance()
     if (is<CDMInstanceOpenCDM>(*m_cdmInstance)) {
         GST_DEBUG("handling OpenCDM %s keys", m_cdmInstance->keySystem().utf8().data());
         auto& cdmInstanceOpenCDM = downcast<CDMInstanceOpenCDM>(*m_cdmInstance);
-        String sessionId = cdmInstanceOpenCDM.getCurrentSessionId();
-        ASSERT(!sessionId.isEmpty());
-        dispatchDecryptionSession(sessionId);
+        size_t index = 0;
+        // Optimized this in case there's only one init data to avoid comparisons and because in some cases,
+        // the JS app send us down a new init data we can't map (in this case it would be a FIXME for the case
+        // it has more than one unmapped init datas).
+        LockHolder lock(m_protectionMutex);
```

Comment:
Better to put RAII locks in a new scope rather than rely on calling unlockEarly in the right place